libraries/AP_HAL_ChibiOS/hwdef/ACNS-CM4pilot/hwdef.dat

```diff
@@ -0,0 +1,215 @@
+PC15 ESPI_CS CS
+
+# I2C1
+PB6 I2C1_SCL I2C1 
+PB7 I2C1_SDA I2C1 
+
+#define HAL_USE_I2C TRUE
+#define STM32_I2C_USE_I2C1 TRUE
+# I2C Buses
+I2C_ORDER I2C1 
+#define HAL_I2C_CLEAR_ON_TIMEOUT 0
+
+# CAN
+PB8 CAN1_RX CAN1
+PB9 CAN1_TX CAN1
+# CAN Buses
+#CAN_ORDER 1 
+
+# reduce memory usage
+define UAVCAN_NODE_POOL_SIZE 1024
+
+PC0 RSSI_ADC_PIN ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+PC2 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+
+PC14 LED_BLUE OUTPUT LOW GPIO(0)
+PA13 LED_GREEN OUTPUT LOW GPIO(1) 
+#PC15 LED_RED OUTPUT LOW GPIO(2) 
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+#define HAL_GPIO_C_LED_PIN 2
+
+
+#pwm output
+PB0  TIM3_CH3  TIM3 PWM(1) GPIO(50)
+PB5  TIM3_CH2  TIM3 PWM(2) GPIO(51)
+PB4  TIM3_CH1  TIM3 PWM(3) GPIO(52)
+PB1  TIM3_CH4  TIM3 PWM(4) GPIO(53)
+PB3  TIM2_CH2  TIM2 PWM(5) GPIO(54)
+PA15 TIM2_CH1  TIM2 PWM(6) GPIO(55)
+PC8  TIM8_CH3  TIM8 PWM(7) GPIO(56)
+PC9  TIM8_CH4  TIM8 PWM(8) GPIO(57)
+#PWM output for buzzer
+PA8  TIM1_CH1  TIM1 GPIO(77) LOW ALARM
+#PA2  TIM2_CH3  TIM2 PWM(10) GPIO(59) 
+
+PC13 VBUS INPUT OPENDRAIN
+
+# default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+#################################################
+###                 DEVICES                   ###
+#################################################
+
+# BMI088 on SPI1
+SPIDEV bmi088_g SPI1 DEVID1  BMI088_G_CS  MODE3 10*MHZ 10*MHZ
+SPIDEV bmi088_a SPI1 DEVID2  BMI088_A_CS  MODE3 10*MHZ 10*MHZ
+
+# MAG on SPI2
+SPIDEV lis3mdl SPI2 DEVID3 MAG_CS MODE3 500*KHZ 500*KHZ
+
+# SD Card on SPI3
+SPIDEV sdcard SPI3 DEVID4 SDCARD_CS MODE3 400*KHZ 25*MHZ
+
+# Flash Card on SPI3
+SPIDEV dataflash SPI3 DEVID5 FLASH_CS MODE3 500*KHZ 25*MHZ
+
+# one IMU
+IMU BMI088 SPI:bmi088_a SPI:bmi088_g ROTATION_ROLL_180_YAW_90
+#ROTATION_ROLL_180_YAW_270
+
+# one baro, multiple possible choices for different board variants
+BARO BMP280 I2C:0:0x76
+
+# one mag
+COMPASS LIS3MDL SPI:lis3mdl false ROTATION_ROLL_180_YAW_270
+
+
+define HAL_OS_FATFS_IO 1
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+
+# define default battery setup
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 12
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 10.1
+define HAL_BATT_CURR_SCALE 17.0
+
+#analog rssi pin (also could be used as analog airspeed input)
+# PC0 - ADC1_CH10
+define BOARD_RSSI_ANA_PIN 10
+
+# probe the i2c bus for all possible
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+
+#define OSD_ENABLED 0
+
+define STM32_PWM_USE_ADVANCED TRUE
+
+
+
+# --------------------- save flash ----------------------
+include ../include/minimize_features.inc
+# only include ublox GPS driver
+#include ../include/minimal_GPS.inc
+
+# reduce max size of embedded params for apj_tool.py
+define AP_PARAM_MAX_EMBEDDED_PARAM 1024
+#define HAL_GENERATOR_ENABLED 0
```

Comment:
I'm guessing these have been swiped from somewhere else.  Are these really appropriate for the board you are producing?

Perhaps including "save_some_flash.inc" would be better.

Reply:
I did learn to write this file from another hwdef file, and the compiled code is running fine on our hardware.
We will test "save_some_flash.inc" Thank you for your suggestion